torch/distributed/_composable/_ddp.py

```diff
@@ -1166,6 +1167,15 @@ def forward(self, *inputs, **kwargs):
             )
         return output
 
+    def forward(self, *inputs, **kwargs):
+        self.pre_forward(*inputs, **kwargs)
+        with torch.autograd.profiler.record_function(
```

Comment:
this is not the same as existing DDP? As the `record_function` used to wrap pre_forward and post_forward here as well?

Reply:
That's true, it's not exactly the same.
Since it's now broken into 2 other functions, I added `record_function("DistributedDataParallel.pre_forward")` and `record_function("DistributedDataParallel.post_forward")` in each of them. Does this make sense?